compiler/src/iree/compiler/Codegen/Common/IREEExpandStridedMetadata.cpp

```diff
@@ -138,8 +139,6 @@ struct ResolveExtractMetadataFromHalInterfaceBindingSubspan
     if (!binding)
       return failure();
     auto memRefType = llvm::cast<MemRefType>(binding.getResult().getType());
-    if (memRefType.getRank() < 1)
```

Comment:
Don't we need to avoid memrer of zero ranks? There is nothing to be done with those

Reply:
We still need to resolve the metadata, and there might still be an offset

That is to say, things break if I didn't delete this

Reply:
That is, we still need to return the offset (if there is one), push the base pointer through, etc.

I suspect we don't need to create a new 1D binding, but ... maybe we do, if it's a 0-D with an offset.

Maybe what we need is logic to allow for `memref<T, strided<[], offset=?>`, but we might as well make that a `memref<1xT, strided=<[1], offset =?>` for uniformity's sake and to keep size bounds

Reply:
I am just thinking this might go into an infinite loop. Nothing changes if the memref is already 0d right? So then this creates a new op of 0D and replaces the old op of 0d and so on?

Reply:
No, it doesn't. First, we only create a new binding if the 0D memref has an offset. Second, this rewrite only triggers when there's an `extract_strided_slice` operation pointed at the binding, and that operation is always RAUW'd at the end of the pattern